test/pkg/compose/build/build_blueprint.go

```diff
@@ -0,0 +1,279 @@
+				klog.ErrorS(err, "Composing commit failed", "blueprint", b.Name)
+				return err
+			}
+			return nil
+		})
+	}
+
+	if b.Installer {
+		if isoExists, err := util.PathExistsAndIsNotEmpty(b.InstallerDestination); err != nil {
+			return err
+		} else if isoExists && !opts.Force {
+			klog.InfoS("ISO installer already present in vm-storage and --force wasn't present - skipping", "blueprint", b.Name)
+		} else {
+			eg.Go(func() error {
+				// TODO: Retry
+				if err := b.composeInstaller(opts); err != nil {
+					klog.ErrorS(err, "Composing installer failed", "blueprint", b.Name)
+					return err
+				}
+				return nil
+			})
+		}
+	}
+
+	err = eg.Wait()
+	if err != nil {
+		return err
+	}
+
+	if !skipCommit && len(b.Aliases) != 0 {
+		klog.InfoS("Adding aliases", "name", b.Name)
+		err = opts.Ostree.CreateAlias(b.Name, b.Aliases...)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (b *BlueprintBuild) composeCommit(opts *Opts) error {
+	var commitID string
+	var err error
+
+	start := time.Now()
+	klog.InfoS("Starting commit compose procedure", "blueprint", b.Name)
+
+	commitID, err = opts.Composer.StartOSTreeCompose(b.Name, "edge-commit", b.Name, b.Parent)
+	if err != nil {
+		return err
+	}
+
+	friendlyName := fmt.Sprintf("%s_edge-commit_%s", b.Name, commitID)
+
+	waitErr := opts.Composer.WaitForCompose(commitID, friendlyName, 15*time.Minute)
+
+	// Get metadata and logs even if composing failed
+	metadataErr := opts.Composer.SaveComposeMetadata(commitID, friendlyName)
+	logsErr := opts.Composer.SaveComposeLogs(commitID, friendlyName)
+
+	if err := errors.Join(waitErr, metadataErr, logsErr); err != nil {
+		return err
+	}
+
+	commitArchivePath, err := opts.Composer.SaveComposeImage(commitID, friendlyName, ".tar")
+	if err != nil {
+		return err
+	}
+	err = opts.Ostree.ExtractCommit(commitArchivePath)
+	if err != nil {
+		return err
+	}
+
+	klog.InfoS("Commit compose procedure done", "blueprint", b.Name, "elapsed", time.Since(start))
+
+	return nil
+}
+
+func (b *BlueprintBuild) composeInstaller(opts *Opts) error {
+	start := time.Now()
+	klog.InfoS("Starting installer compose procedure", "blueprint", b.Name)
+
+	installerID, err := opts.Composer.StartCompose(b.Name, "image-installer")
+	if err != nil {
+		return err
+	}
+
+	friendlyName := fmt.Sprintf("%s_image-installer_%s", b.Name, installerID)
+
+	waitErr := opts.Composer.WaitForCompose(installerID, friendlyName, 25*time.Minute)
+
+	// Get metadata and logs even if composing failed
+	metadataErr := opts.Composer.SaveComposeMetadata(installerID, friendlyName)
+	logsErr := opts.Composer.SaveComposeLogs(installerID, friendlyName)
+
+	if err := errors.Join(waitErr, metadataErr, logsErr); err != nil {
+		return err
+	}
+
+	installerPath, err := opts.Composer.SaveComposeImage(installerID, friendlyName, ".iso")
+	if err != nil {
+		return err
+	}
+
+	dest := filepath.Join(opts.ArtifactsMainDir, "vm-storage", b.Name+".iso")
+	if installerPath != "/dummy/dry/run/path" {
```

Comment:
What's this?

Reply:
This is making sure that this function doesn't try to move filepath that comes from `composerDryRun`. I'm thinking about moving it to the `Composer` interface so dryrun would simply ignore it